torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -555,6 +555,19 @@ def ReplaceMoreTtManually(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
                     permutation = [1, 0]
                     new_node = g.call_function(ttnn.permute, args=(args[0], permutation))
                     new_nodes.append(new_node)
+            elif node.target == torch.ops.aten.max_pool2d_with_indices.default:
+                operand = args[0]
+                operand_shape = operand.meta["val"].size()
+                kwargs = {
+                    "in_n": operand_shape[0],
+                    "in_h": operand_shape[2],
+                    "in_w": operand_shape[3],
+                    "kernel_h": args[1][0],
+                    "kernel_w": args[1][1],
+                    "return_indices": True,
+                    **node.kwargs,
+                }
+                new_node = g.call_function(ttnn.max_pool2d, (operand,), kwargs)
```

Comment:
I wrote the conversion here, but it does not take effect
```
---------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
opcode         name                             target                                args                                  kwargs
-------------  -------------------------------  ------------------------------------  ------------------------------------  --------
placeholder    arg0_1                           arg0_1                                ()                                    {}
call_function  max_pool2d_with_indices_default  aten.max_pool2d_with_indices.default  (arg0_1, [3, 3])                      {}
call_function  getitem                          <built-in function getitem>           (max_pool2d_with_indices_default, 0)  {}
call_function  getitem_1                        <built-in function getitem>           (max_pool2d_with_indices_default, 1)  {}
output         output                           output                                ((getitem, getitem_1),)               {}
-------------------------------------------------------------------------- Captured stdout teardown --------------------------------------------------------------------------
                  Metal | INFO     | Closing device 0
                  Metal | INFO     | Disabling and clearing program cache on device 0
========================================================================== short test summary info ===========================================================================
FAILED tests/lowering/pool/test_max_pool_2d.py::test_max_pool_2d_plain[input_shapes0-kernel_size0] - AssertionError: assert 0 == 1
FAILED tests/lowering/pool/test_max_pool_2d.py::test_max_pool_2d_with_indices[input_shapes0-kernel_size0] - AssertionError: assert 0 == 1
```

However, I found an interesting result.  Even in the case where `return_indices=False`, i.e. not producing the "argmax" tensor, PyTorch still generates `aten.max_pool2d_with_indices_default` instead of `aten.max_pool2d_native`.  I doubt if MNIST uses the argmax result at all. :monocle_face:
```
---------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
opcode         name                             target                                args                                  kwargs
-------------  -------------------------------  ------------------------------------  ------------------------------------  --------
placeholder    arg0_1                           arg0_1                                ()                                    {}
call_function  max_pool2d_with_indices_default  aten.max_pool2d_with_indices.default  (arg0_1, [3, 3])                      {}
call_function  getitem                          <built-in function getitem>           (max_pool2d_with_indices_default, 0)  {}
output         output                           output                                ((getitem,),)                         {}
```

Reply:
@jdh8 this feature is simply not supported. 
Can you simply lower to conv2d and fire a ticket mentioning that the case where this param is True is not supported?

Reply:
I fired tenstorrent/tt-metal#12099 that requests support for `aten.max_pool2d_with_indices.default`.  For the workaround, I still need tech support from @kevinwuTT to lower such a nested expression:
```py
aten.max_pool2d_with_indices.default(x)[0]
```

Reply:
We need to rewrite a sequence of two ops to a single new op.

From:
```py
intermediate = aten.max_pool2d_with_indices.default(x)
result = intermediate[0]
```

To:
```py
result = ttnn.max_pool2d(x)
```

Reply:
Layer_norm conversion had this similar issue with getitem. For models that don't use backwards propagation, index 0 is enough. This might not be the cleanest approach, but I have some logic for handling layer_norm here. https://github.com/tenstorrent/pytorch2.0_ttnn/blob/9277f6ff179deac536f1e4ccefd390ff485a4ec5/torch_ttnn/passes/lowering/add_data_move_pass.py#L363